### PR TITLE
Add 2023 Connecticut Tax Rules

### LINKED ISSUE #4170: Add 2023 Connecticut Tax Rules
2023 Income Tax Form Instructions: https://portal.ct.gov/-/media/DRS/Forms/2023/Income/2023-CT-1040-Instructions_1223.pdf

### PR DESCRIPTION
**Pull request by vrathi101**
Fixes #4170

Changes:
- CT EITC match percentage
- Reduction in certain income tax rates for 2024

### CODE REVIEW

File: policyengine_us/parameters/gov/states/ct/tax/income/rates/head_of_household.yaml [hunk 1]
```diff
@@ -7,6 +7,8 @@ metadata:
   reference:
     - title: Connecticut General Statutes, Chapter 229, Sec. 12-700, (9), (B)
       href: https://www.cga.ct.gov/current/pub/chap_229.htm#sec_12-700 
+    - title: 2023 Form CT-1040 Connecticut Resident Income Tax Return Instructions
+      href: https://portal.ct.gov/-/media/DRS/Forms/2023/Income/2023-CT-1040-Instructions_1223.pdf#page=20
     - title: 2022 Form CT-1040 Connecticut Resident Income Tax Return Instructions
       href: https://portal.ct.gov/-/media/DRS/Forms/2022/Income/2022-CT-1040-Instructions_1222.pdf#page=20
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ct/tax/income/rates/head_of_household.yaml:13]: could you take a look here:
https://portal.ct.gov/DRS/Miscellaneous-Taxes/Other-Tax-Page/State-Tax-Developments#:~:text=Specifically%2C%20the%203%25%20rate%20on,couples%20will%20decrease%20to%204.5%25.

In 2023 the rate is the same but in 2024 there are changes, could add this as well while we are here

File: policyengine_us/parameters/gov/states/ct/tax/income/subtractions/pensions_or_annuity/income_limit.yaml [hunk 2]
```diff
@@ -8,6 +8,8 @@ metadata:
   reference:
     - title: Connecticut General Statutes, Chapter 229, (20), (B), (xxi) 
       href: https://www.cga.ct.gov/current/pub/chap_229.htm#sec_12-701a
+    - title: 2023 Form CT-1040 Connecticut Resident Income Tax Return Instructions
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ct/tax/income/subtractions/pensions_or_annuity/income_limit.yaml:11]: The same [source](https://portal.ct.gov/DRS/Miscellaneous-Taxes/Other-Tax-Page/State-Tax-Developments#:~:text=Specifically%2C%20the%203%25%20rate%20on,couples%20will%20decrease%20to%204.5%25.) as above also talks about a phase-out here, is there anything in the forms?

image

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/parameters/gov/states/ct/tax/income/subtractions/pensions_or_annuity/income_limit.yaml (hunk 2 above)
Comment (vrathi101) [reply policyengine_us/parameters/gov/states/ct/tax/income/subtractions/pensions_or_annuity/income_limit.yaml:11]: Looked through the form but did not see this mentioned for the pension/annuity section

Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/ct/tax/income/subtractions/pensions_or_annuity/income_limit.yaml:11]: not in the tax form either?

**Review by PavelMakarchuk — APPROVED** [review]

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**